analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -70,6 +71,61 @@ def parse_analyzer_config(config_dump):
     return re.findall(r'^(\S+):\s+(\S+)$', config_dump, re.MULTILINE)
 
 
+def get_diagtool_bin():
+    """
+    Return full path of diagtool.
+
+    Select clang binary, check for a 'diagtool' binary next to the selected
+    clang binary and return full path of this binary if it exists.
+    """
+    context = analyzer_context.get_context()
+    clang_bin = context.analyzer_binaries.get(ClangSA.ANALYZER_NAME)
+
+    if not clang_bin:
+        return None
+
+    # Resolve symlink.
+    clang_bin = os.path.realpath(clang_bin)
+
+    # Find diagtool next to the clang binary.
+    diagtool_bin = os.path.join(os.path.dirname(clang_bin), 'diagtool')
+    if os.path.exists(diagtool_bin):
+        return diagtool_bin
+
+    LOG.debug("'diagtool' can not be found next to the clang binary (%s)!",
+              clang_bin)
+
+
+def get_warnings(env=None):
+    """
+    Returns list of warning flags by using diagtool.
+    """
+    diagtool_bin = get_diagtool_bin()
+
+    if not diagtool_bin:
+        return []
+
+    try:
+        result = subprocess.check_output(
+            [diagtool_bin, 'tree'],
+            env=env,
+            universal_newlines=True,
+            encoding="utf-8",
+            errors="ignore")
+        return [w[2:] for w in result.split()
+                if w.startswith("-W") and w != "-W"]
```

Comment:
This is a good point. However, at this point we don't omit `-W` because of it's aggregated, but because in the previous line we take a sub-string from the 2nd character.
By the way yes, we could consider handling `-Wall` and `-Wextra` separately. In the future we'll expect the user to use the `clang-diagnostic-` version of these warnings. If `clang-diagnostic-all` and `clang-diagnostic-extra` are not supported, then those should be excluded. Fun fact: I've just noticed, there is a `-Wmost` in this list :)